images/metering-ansible-operator/roles/meteringconfig/defaults/main.yml

```diff
@@ -120,6 +161,28 @@ _storage_overrides:
         hdfs:
           enabled: false
 
+# overides the accountname(potentially empty) from a designated secret and generate the storage URL
+meteringconfig_storage_azure_create_secret: "{{ _storage_spec | json_query('hive.azure.createSecret') }}"
+
+_azure_secret_account_name: ""
+
+_azure_account_overides:
+  azure:
+
+    hadoop:
+      spec:
+        config:
+          defaultFS: "wasbs://{{ _storage_spec | json_query('hive.azure.container') }}@{{ _azure_secret_account_name  }}.blob.core.windows.net{{ _storage_spec | json_query('hive.azure.rootDirectory')| default('/', True) | regex_replace('^\\/?', '/') }}"
+    hive:
+      spec:
+        config:
+          metastoreWarehouseDir: "wasbs://{{ _storage_spec | json_query('hive.azure.container') }}@{{ _azure_secret_account_name }}.blob.core.windows.net{{ _storage_spec | json_query('hive.azure.rootDirectory')| default('/', True) | regex_replace('^\\/?', '/') }}"
+
+_meteringconfig_azure_overrides: "{{ meteringconfig_storage_azure_create_secret | ternary({} , _azure_account_overides ) | default({}, true) }}"
```

Comment:
I don't feel like `meteringconfig_storage_azure_create_secret` is the correct condition here since it will be false when users provide an existing secret. I think you need to check if `storage.hive.azure` is set.

Reply:
Ah I know. Check `meteringconfig_storage_hive_storage_type`, but thats' defined below, so you might need to re-organize a bit.

Reply:
> Ah I know. Check `meteringconfig_storage_hive_storage_type`, but that's defined below, so you might need to re-organize variable.

to my understanding I think its okay to generate a bogus value here because the hive type selector will select the correct value later on, it will only overwrite the 's3' part if its s3 for example. we don't want to select base on type because it will overwrite accountName to null if createSecret is true etc?

Reply:
Let's add a comment saying:
```
# Use _azure_account_overides when meteringconfig_storage_azure_create_secret is false to correctly set the defaultFS and metastoreWarehouseDir with the storage account name from the user's secret.
```